modules/terraform/init_test.go

```diff
@@ -35,3 +37,43 @@ func TestInitBackendConfig(t *testing.T) {
 
 	assert.FileExists(t, remoteStateFile)
 }
+
+func TestInitPluginDir(t *testing.T) {
+	t.Parallel()
+
+	pluginDir, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+
+	terraformFixture := "../../test/fixtures/terraform-basic-configuration"
+
+	initializePluginsFolder, err := files.CopyTerraformFolderToTemp(terraformFixture, t.Name())
+	require.NoError(t, err)
+
+	testFolder, err := files.CopyTerraformFolderToTemp(terraformFixture, t.Name())
+	require.NoError(t, err)
+
+	defer os.RemoveAll(initializePluginsFolder)
+	defer os.RemoveAll(testFolder)
+	defer os.RemoveAll(pluginDir)
```

Comment:
These defers should be done immediately after each dir is created. That way, the clean up happens even if the subsequent temp folders fail to create.

For example, as written, it won't cleanup `pluginDir` if there is an error creating the `initializePluginsFolder`.

Note that each call should be done after the `require.NoError` call so that we only schedule cleaning up if it was successfully created.

Reply:
ahh yeah whoops!